src/service/service-agreement-service.js

```diff
@@ -11,10 +9,11 @@ class ServiceAgreementService {
         this.networkModuleManager = ctx.networkModuleManager;
     }
 
-    async generateId(assetTypeContract, tokenId, keyword, hashFunctionId) {
+    async generateId(blockchain, assetTypeContract, tokenId, keyword, hashFunctionId) {
         return this.validationModuleManager.callHashFunction(
             hashFunctionId,
-            ethers.utils.solidityPack(
+            this.blockchainModuleManager.encodePacked(
```

Comment:
this type of functions should also be moved somewhere else since they don't depend on blockchain implementation